### PR TITLE
Issue #1602 Remove vmware-fusion mention/support from documentation

### PR DESCRIPTION
**Pull request by athos-ribeiro**
Refers to #1602

### CODE REVIEW

**Comment by gbraad** [comment]
```
FAIL - commit message does not match "(Issue #[0-9]+ .*|cut v[0-9]+.[0-9]+.[0-9]+)"
```

Commit message should refer to the issue number as follows: "Issue #1602 Remove ..."

**Comment by praveenkumar** [comment]
There are also other places in the doc and also in the code which should be taken care as part of this PR.

Doc section

+ `/docs/source/getting-started/installing.adoc:31:- link:https://www.vmware.com/products/fusion[VMware Fusion]`

Code section

+ https://github.com/minishift/minishift/blob/master/pkg/minikube/cluster/cluster.go#L395-L412
+ https://github.com/minishift/minishift/blob/master/pkg/minikube/machine/client.go#L37-L38

**Comment by budhrg** [comment]
retest this please

**Comment by budhrg** [comment]
> Should they aiso be removed?

@athos-ribeiro Yes please

**Comment by athos-ribeiro** [comment]
Done

**Comment by LalatenduMohanty** [comment]
@minishift/minishift-dev -1 from my side for removing the code related to VMware Fusion. Because we do not want to break the setup of any users who is using VMware fusion or in future if we decide to add the support. I think removing documentation is enough for indicating that it is not supported. Also this issue was only for removing the documentation.

**Comment by gbraad** [comment]
I am for removing the code, but in the sense of a deprecation period. We already kinda mention we do not support the setup when they report an issue involving VMware. At the Minikube side there is:

  * https://github.com/kubernetes/minikube/issues/2058 to drop support

Unless a maintained driver is available the current support is nothing more than flaky. The driver hasn't seen much development. The VMware repo of libmachine has not seen improvements for 2 years on their master: https://github.com/vmware/docker-machine/tree/master/drivers/vmwarefusion. The same for the official repo of libmachine https://github.com/docker/machine/tree/master/drivers/vmwarefusion which has only seen typo changes. Either the code is super-stable... or no-one provides support. The same for the minikube repo: https://github.com/kubernetes/minikube/tree/master/vendor/github.com/docker/machine/drivers/vmwarefusion they synced a year ago and have since not provide any fixes.

There is ongoing work

  * https://github.com/kubernetes/minikube/pull/2118

which is a vmware fusion and workstation driver, but this will not be a drop-in driver for us, as there is likely no RPC support, which we would still need (as we use libmachine unmodified, while minikube has an in-repo modification).

**Comment by gbraad** [comment]
@minishift/minishift-dev Also important to note is that libmachine has no intention to add more functionality: https://github.com/docker/machine/pull/4280#issuecomment-338842944 such as an updated driver.

**Comment by gbraad** [comment]
We will do the code changes in a different PR when we have agreed...

**Comment by LalatenduMohanty** [comment]
retest this please

**Comment by budhrg** [comment]
> I am for removing the code, but in the sense of a deprecation period. We already kinda mention we do not support the setup when they report an issue involving VMware. At the Minikube side there is:

I am too in favour of removing code but certainly not the good idea to remove it completely.
Let's go with deprication.

Depreciation message should goes here as part of of 1.9.1.

https://github.com/minishift/minishift/blob/c677a9853b13c96e6353bacd809da5648d20375f/pkg/minikube/cluster/cluster.go#L396-L412

**Comment by gbraad** [comment]
@budhrg @LalatenduMohanty I am OK with moving the removal back. So this remains a documentation issue... By adding a deprecation notice we can also avoid having to document this further in the documentation.

I will keep an eye on the Minikube development around the driver and see what @anfernee might might do. An alternative approach would be to create an external driver, much like KVM and xhyve has. This way we can avoid direct support.

**Comment by budhrg** [comment]
@athos-ribeiro , could you add the deprecation message 
"VMWare fusion driver is deprecating soon. Please consider using other drivers."

above this line 

https://github.com/minishift/minishift/blob/master/pkg/minikube/cluster/cluster.go#L396

and remove all code changes keeping documentation changes.

/cc @LalatenduMohanty @gbraad

**Comment by gbraad** [comment]
@athos-ribeiro sorry about us going slightly back and forth, but this is because of our userbase and the driver proposal that exists at minikube

**Comment by athos-ribeiro** [comment]
> @athos-ribeiro , could you add the deprecation message

done. I wasn't sure if I should just print the message or log it. I went with the former, but if you prefer otherwise, just let me know.

> and remove all code changes keeping documentation changes.

done

> @athos-ribeiro sorry about us going slightly back and forth, but this is because of our userbase and the driver proposal that exists at minikube

I understand things must be done the right way :) Thanks for the patience guiding me through this PR.

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -393,6 +393,7 @@ func getDriverOptions(config MachineConfig) (drivers.DriverOptions, error) {
 
 		return createDriverOptions(d, machineConfigOptions)
 	case "vmwarefusion":
+		fmt.Println("VMWare fusion driver is deprecating soon. Please consider using other drivers.")
```
Comment (robin-owen) [inline pkg/minikube/cluster/cluster.go]: VMware Fusion is improperly capitalized.

Personally, I would also prefer "will be deprecated" to "is deprecating".

File: docs/source/getting-started/setting-up-driver-plugin.adoc [hunk 2]
```diff
@@ -12,7 +12,7 @@ toc::[]
 [[setup-driver-plugin-overview]]
 == Overview
 {project} uses Docker Machine and its driver plug-in architecture to provide a consistent way to manage the OpenShift VM.
-{project} embeds VirtualBox and VMware Fusion drivers so no additional steps are required to use them.
+{project} embeds VirtualBox driver so no additional steps are required to use it.
```
Comment (robin-owen) [inline docs/source/getting-started/setting-up-driver-plugin.adoc]: This feels like cleaner writing to me:

```adoc
{project} embeds a VirtualBox driver.
No additional steps are required to use it.
```

At the very least, it should be "a VirtualBox driver".

Comment (LalatenduMohanty) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: @robin-owen a VirtualBox driver" does not sound correct to me as there is only one implementation for the Virtualbox driver. So it should be "the VirtualBox driver".

**Comment by budhrg** [comment]
add to whitelist

File: docs/source/getting-started/setting-up-driver-plugin.adoc (hunk 2 above)
Comment (gbraad) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: To be nitpicking in that case, it is called a 'driver plugin' in all the documentation and references at docker/machine. This is also what we use in our configuration topic.

Comment (robin-owen) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: > "a VirtualBox driver" does not sound correct to me as there is only one implementation for the Virtualbox driver. So it should be "the VirtualBox driver".

I wasn't sure if there was more than one implementation -- in this case, yes, "the" makes more sense.

> To be nitpicking

This isn't nitpicking if it's more technically accurate to call it "the VirtualBox driver plug-in".

**Comment by gbraad** [comment]
Hope we can merge this soon.

  * [ ] "will be deprecated"
  * [ ] "the VirtualBox driver plugin"

**Comment by praveenkumar** [comment]
@athos-ribeiro ping, Can you please update your PR as per suggestion so we can merge it asap?

**Comment by athos-ribeiro** [comment]
- Fixed the strings as requested
- Rebased against master

**Comment by praveenkumar** [comment]
@LalatenduMohanty changes are already made by @athos-ribeiro, can we merge it now?

**Comment by LalatenduMohanty** [comment]
@athos-ribeiro Thanks for the fix.

**Pull request merged**